src/primitives/sphere.jl

```diff
@@ -37,6 +37,38 @@ end
 
 Sphere(p1::Tuple, p2::Tuple, p3::Tuple) = Sphere(Point(p1), Point(p2), Point(p3))
 
+"""
+    Sphere(p1, p2, p3, p4)
+
+A 3D sphere passing through points `p1`, `p2`, `p3` and `p4`.
+"""
+function Sphere(p1::Point{3}, p2::Point{3}, p3::Point{3}, p4::Point{3})
+  x1 = coordinates(p1)
+  x2 = coordinates(p2)
+  x3 = coordinates(p3)
+  x4 = coordinates(p4)
+  X = hcat(x1, x2, x3, x4)
+  a = 2 * det(vcat(X, ones((1, 4))))
+  T = typeof(a)
```

Comment:
You are trying to compute the volume of the prism right? We prefer to be explicit in our code, so try to rewrite this as volume(Tetrahedron(p1, p2, p3, p4))

Reply:
Clearly this determinant is zero if and only if the the volume of the tetrahedron is zero, but I don't know whether these are the same things.

Reply:
From a quick look, the volume you are computing here is an integer multiple of the volume of the tetrahedron. What I am saying is that it is useful to write algorithms in terms of high-level functions to increase coverage and to avoid fixing a bug in an isolated place.